### ISSUE TITLE
Confusing error message for failed positive lookahead expression

### ISSUE THREAD

**Issue by vsemozhetbyt** [pegjs/pegjs#640]
### Issue type

- **Bug Report:** _yes_ 
- **Feature Request:** _maybe_
- **Question:** _no_
- **Not an issue:** _no_

### Prerequisites

- **Can you reproduce the issue?:** _yes_ 
- **Did you search the repository issues?:** _yes_ 
- **Did you check the forums?:** _no_ (cannot find the URL)
- **Did you perform a web search (google, yahoo, etc)?:** _no_ (seems unneeded)

### Description

Error message for failed positive lookahead expression does not mention this expression

### Steps to Reproduce

Consider a rule where double hyphens are not allowed:
```js
const pegResult = require('pegjs/')
  .generate('root = ([^-] / "-" !"-")+')
  .parse('a-b--c');

console.log(pegResult);
```

**Expected behavior:**

Something like: `peg$SyntaxError: Expected "-" !"-", [^\-], or end of input but "--" found.`

**Actual behavior:**

`peg$SyntaxError: Expected "-", [^\-], or end of input but "-" found.`

### Software

- **PEG.js:** `0.11.0-master.b7b87ea`
- **Node.js:** `14.0.0-v8-canary202002123546737a75`
- **NPM or Yarn:** `npm 6.13.7`
- **Browser:** -
- **OS:** Windows 7 x64
- **Editor:** -

**Comment by StoneCypher** [comment]
Hi, I'm not very good at `peg` yet.  I want to make sure I'm understanding this correctly.

It seems like what you're saying is "I have a rule where `abcd` is forbidden because `d` is disallowed as a lookahead in a rule about `cd`.  When it complains, I'd prefer to hear about `cd` but I only hear about `d`."

Is that roughly correct?

**Comment by vsemozhetbyt** [comment]
I would rather say: "I have a rule where `abcc` is forbidden because `c` is disallowed as a lookahead for a `c` in a rule about `abc`. When it complains, I'd prefer to hear about `cc `but I only hear about `c`." Otherwise, the error message is not useful or clear: Expected "c"... but "c" found.

**Comment by StoneCypher** [comment]
Oh, the error message is confusing ***because*** the forbiddance is a repetition?

**Comment by vsemozhetbyt** [comment]
I would say the message is confusing because

1) `Expected` and `Found` is the same in this case and
2) the real cause of failure is not clear.

Maybe it would be useful to mention lookahead expressions if they are the cause of failure. Forbidden repetitions are just a case, compare this without a forbidden repetition.

```js
const pegResult = require('pegjs/')
  .generate('root = ("a" / "b" !"a")+')
  .parse('aba');

console.log(pegResult);
```

```
peg$SyntaxError: Expected "a", "b", or end of input but "b" found.
```

**Comment by vsemozhetbyt** [comment]
Sorry, I am a bit clumsy, as I've just started to learn the PEG.

BTW, I've tried about dozen other parser generators (I am a newbie in this area) and this seems to be my favorite now)

**Comment by StoneCypher** [comment]
Yeah, peg's almost the only parser generator for `javascript` that I consider acceptable

Don't worry about being clumsy.  We all are

**Comment by Mingun** [comment]
@vsemozhetbyt, actually, you grammar should issue a info/warning at build stage, because it accept only inputs, which are accepted by grammar
```peg
root
  = "a"+ "b"*
  / "b"+
```
but compiler not (yet?) smart enough to infer that. I even don't know, may be it is NP-complete problem...

**Comment by StoneCypher** [comment]
the thing about np complete problems is that you can often still do a pretty decent job heuristically

if it's often right and never wrong, then it's a defacto improvement
